pfsimulator/parflow_exe/main.c

```diff
@@ -113,6 +118,66 @@ int main(int argc, char *argv [])
     cegdb(&argc, &argv, amps_Rank(MPI_CommWorld));
 #endif
 
+#if ACC_BACKEND == BACKEND_CUDA
+
+#ifndef NDEBUG
+    /*-----------------------------------------------------------------------
+    * Wait for debugger if MPI_DEBUG_RANK environment variable is set
+    *-----------------------------------------------------------------------*/
+    if(getenv("MPI_DEBUG_RANK") != NULL) {
+      const int mpi_debug = atoi(getenv("MPI_DEBUG_RANK"));
+      if(mpi_debug == amps_Rank(amps_CommWorld)){
+        volatile int i = 0;
+        amps_Printf("MPI_DEBUG_RANK environment variable found.\n");
+        amps_Printf("Attach debugger to PID %ld (MPI rank %d) and set var i = 1 to continue\n", (long)getpid(), mpi_debug);
+        while(i == 0) {/*  change 'i' in the  debugger  */}
+      }
+      amps_Sync(amps_CommWorld);
+    }
+#endif // !NDEBUG
+
+    /*-----------------------------------------------------------------------
+    * Check CUDA compute capability, set device, and initialize RMM allocator
+    *-----------------------------------------------------------------------*/
+    {
+      // CUDA
+      if (!amps_Rank(amps_CommWorld))
+      {
+        CUDA_ERR(cudaSetDevice(0));  
+      }else{
+        int num_devices = 0;
+        CUDA_ERR(cudaGetDeviceCount(&num_devices));
+        CUDA_ERR(cudaSetDevice(amps_node_rank % num_devices));
+      }
+    
+      int device;
+      CUDA_ERR(cudaGetDevice(&device));
+
+      struct cudaDeviceProp props;
+      CUDA_ERR(cudaGetDeviceProperties(&props, device));
+
+      // int value;
+      // CUDA_ERR(cudaDeviceGetAttribute(&value, cudaDevAttrCanUseHostPointerForRegisteredMem, device));
+      // printf("cudaDevAttrCanUseHostPointerForRegisteredMem: %d\n", value);
+
+      if (props.major < 6)
+      {
+        amps_Printf("\nError: The GPU compute capability %d.%d of %s is not sufficient.\n",props.major,props.minor,props.name);
+        amps_Printf("\nThe minimum required GPU compute capability is 6.0.\n");
+        exit(1);
+      }
+
+#ifdef HAVE_RMM
```

Comment:
It can be PARFLOW_HAVE_RMM and it can be moved to the parflow_config file. No PARFLOW_ prefix was initially added to stay consistent with other libraries, e.g., HAVE_CLM, HAVE_OAS3, HAVE_HYPRE, HAVE_SAMRAI, etc.

Reply:
Setting HAVE_RMM in the parflow_config.h makes this #define global across all (?) libraries, e.g., pfsimulator, amps, kinsol, etc. However, RMM must be initialized before it is used, so, for example, the amps tests fail, because HAVE_RMM would be defined but these tests do not initialize RMM. I guess this could be solved by modifying all amps tests such that they initialize RMM if HAVE_RMM is defined? There could be other issues that I do not remember right now.

Reply:
The unprefixed HAVE_CLM etc will be removed.  The config.in file is setup currently defines both but we should be using prefixed symbols.

Reply:
OK with having HAVE_RMM being set then for now since it isn't constant across compilation units.

Could you add a little note in main CMakeLists.txt file with to document the special use.  LIke what you said above... e.g. "Note HAVE_RMM is only set for the compilation of parflow_lib, kinsol sources.   RMM must be initialized before it is used, so, for example, the amps tests would fail if used across all compiles"

So future developers know it is different.
Thanks

Reply:
done